src/components/shared/VirtualList.js

```diff
@@ -2,12 +2,28 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-import React, { PureComponent } from 'react';
+// @flow
+import * as React from 'react';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
 import range from 'array-range';
 
-class VirtualListRow extends PureComponent {
+import type { CssPixels } from '../../types/units';
+
+type RenderItem = (*, number, number) => React.Node;
+
+type VirtualListRowProps = {|
+  +renderItem: RenderItem,
+  +item: *,
```

Comment:
I would rather use a generic so to force-type that `item` and the first argument of `RenderItem` are the same type.

Though I believe that Flow can still find this with the call in `render` so it's probably fine.

Reply:
Yeah not sure what happens either; as I said earlier I think Flow will make it work with the `*` here so let's keep it and maybe refine later.